dvc/repo/plots/__init__.py

```diff
@@ -62,91 +60,90 @@ def collect(
-                    except NoRemoteError:
-                        logger.debug(
-                            (
-                                "Could not find cache for directory '%s' on "
-                                "'%s'. Files inside will not be plotted."
-                            ),
-                            path_info,
-                            rev,
-                        )
-                        continue
-                else:
-                    plot_files = [
-                        (path_info, relpath(path_info, self.repo.root_dir))
-                    ]
-
-                for path, repo_path in plot_files:
-                    data[rev].update({repo_path: {"props": props}})
-
-                    # Load data from git or dvc cache
-                    try:
-                        with fs.open(path) as fd:
-                            data[rev][repo_path]["data"] = fd.read()
-                    except FileNotFoundError:
-                        # This might happen simply because cache is absent
-                        logger.debug(
-                            (
-                                "Could not find '%s' on '%s'. "
-                                "File will not be plotted"
-                            ),
-                            path,
-                            rev,
-                        )
-                    except UnicodeDecodeError:
-                        logger.debug(
-                            (
-                                "'%s' at '%s' is binary file. It will not be "
-                                "plotted."
-                            ),
-                            path,
-                            rev,
-                        )
+            data[rev] = self._collect_from_revision(
+                revision=rev,
+                targets=targets,
+                recursive=recursive,
+                onerror=onerror,
+                props=props,
+            )
 
         return data
 
+    @error_handler
+    def _collect_from_revision(
+        self,
+        targets: Optional[List[str]] = None,
+        revision: Optional[str] = None,
+        recursive: bool = False,
+        onerror: Optional[Callable] = None,
+        props: Optional[Dict] = None,
+    ):
+        from dvc.fs.repo import RepoFileSystem
+
+        fs = RepoFileSystem(self.repo)
+        plots = _collect_plots(self.repo, targets, revision, recursive)
+        res = {}
+        for path_info, rev_props in plots.items():
+
+            if fs.isdir(path_info):
+                plot_files = []
+                for pi in fs.walk_files(path_info):
+                    plot_files.append((pi, relpath(pi, self.repo.root_dir)))
+            else:
+                plot_files = [
+                    (path_info, relpath(path_info, self.repo.root_dir))
+                ]
+
+            props = props or {}
+
+            @error_handler
+            def read(fs, path, props=None, rev_props=None, **kwargs):
```

Comment:
Moved reading plots files here. (Before it was in `to_datapoints`). We are moving parsing here to catch potential parsing error here, and not in `render` level. We could probably get rid of `to_datapoints` method at this point, but I am leaving it as is to not complicate this PR further.  Logic behind `to_datapoints` is strictly related to file structure (eg `csv` is list of datapoints while `json` can be a dict that needs to be explored for data) so it still makes sense to relate data transformation to file extension (which is what `to_datapoints` do now), but it seems to me that we will need to change the approach how we deal with plots data.
Previously it was:
1. read datafiles from revisions
2. parse, transform and render (in `render` method)

Now it is:
1. read and parse datafiles from revsions
2.transform and render

How it will need to be, when we start introducing images support:
1. read and parse (if applicable)
2. transform (if applicable)
3. render